sycl/source/detail/config.def

```diff
@@ -20,3 +20,10 @@ CONFIG(SYCL_DEVICE_FILTER, 1024, __SYCL_DEVICE_FILTER)
 CONFIG(SYCL_PROGRAM_LINK_OPTIONS, 64, __SYCL_PROGRAM_LINK_OPTIONS)
 CONFIG(SYCL_PROGRAM_COMPILE_OPTIONS, 64, __SYCL_PROGRAM_COMPILE_OPTIONS)
 CONFIG(SYCL_HOST_UNIFIED_MEMORY, 1, __SYCL_HOST_UNIFIED_MEMORY)
+CONFIG(SYCL_CACHE_DIR, 1024, __SYCL_CACHE_DIR)
```

Comment:
Does this correlate with Windows and Linux limits on path length?

Reply:
it is ok for Linux but too much for Windows. Will reduce it to 260 (Windows limitation) - 12 (filename) - 84 (cache directory structure) = 172